src/Broadway/EventStore/DBALEventStore.php

```diff
@@ -170,4 +194,30 @@ private function deserializeEvent($row)
             DateTime::fromString($row['recorded_on'])
         );
     }
+
+    private function extractId($id)
+    {
+        if ($this->useBinary) {
+            try {
+                return Uuid::fromString($id)->getBytes();
+            } catch (\Exception $e) {
```

Comment:
@asm89 Well, this is a kind of UUID specific feature, since the BINARY(16) definition matches exactly the uuid v4 byte length. We could also convert i.e. decimals (`dec2bin`) or other numeric values, but I don't see any benifit there.